src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/UpToDate/UpToDateCheckHost.cs

```diff
@@ -9,36 +9,54 @@ namespace Microsoft.VisualStudio.ProjectSystem.VS.UpToDate
     [Export(typeof(IUpToDateCheckHost))]
     internal sealed class UpToDateCheckHost : IUpToDateCheckHost
     {
-        private readonly IVsUIService<SVsShell, IVsShell> _vsShell;
+        private readonly IVsUIService<IVsShell> _vsShell;
+        private readonly IVsUIService<IVsAppCommandLine> _vsAppCommandLine;
         private readonly JoinableTaskContext _joinableTaskContext;
 
         private bool? _hasDesignTimeBuild;
 
         [ImportingConstructor]
-        public UpToDateCheckHost(IVsUIService<SVsShell, IVsShell> vsShell, JoinableTaskContext joinableTaskContext)
+        public UpToDateCheckHost(IVsUIService<SVsShell, IVsShell> vsShell, IVsUIService<SVsAppCommandLine, IVsAppCommandLine> vsAppCommandLine, JoinableTaskContext joinableTaskContext)
         {
             _vsShell = vsShell;
+            _vsAppCommandLine = vsAppCommandLine;
             _joinableTaskContext = joinableTaskContext;
         }
 
         public async ValueTask<bool> HasDesignTimeBuildsAsync(CancellationToken cancellationToken)
         {
-            if (_hasDesignTimeBuild is null)
+            _hasDesignTimeBuild ??= await ComputeValueAsync();
+
+            return _hasDesignTimeBuild.Value;
+
+            async ValueTask<bool> ComputeValueAsync()
             {
                 await _joinableTaskContext.Factory.SwitchToMainThreadAsync(cancellationToken);
 
-                // If VS is running in command line mode, design-time builds do not occur
-                if (ErrorHandler.Succeeded(_vsShell.Value.GetProperty((int)__VSSPROPID.VSSPROPID_IsInCommandLineMode, out object resultObj)) && resultObj is bool result)
+                if (ErrorHandler.Succeeded(_vsAppCommandLine.Value.GetOption("populateSolutionCache", out int populateSolutionCachePresent, out string _)))
                 {
-                    _hasDesignTimeBuild = !result;
+                    if (populateSolutionCachePresent != 0)
```

Comment:
nit: Can merge with outer if. Also in block below.

Reply:
I actually intentionally broke this to add nesting to make it easier to read (subjective I know, but I don't like having the check so far to the right, potentially off screen.)